src/gluonts/model/forecast.py

```diff
@@ -373,7 +373,7 @@ def prediction_length(self):
         """
         Time length of the forecast.
         """
-        return self.samples.shape[-1]
+        return self.samples.shape[1]
```

Comment:
Thank you!

One alternative to consider is to take the `prediction_length` from the `predictor` that construct the forecast. This way, we would be correct (at least in prediction length) even if we change the data layout.